ipldutil/ipldutil.go

```diff
@@ -28,8 +28,11 @@ var (
 	})
 )
 
-func Traverse(ctx context.Context, loader ipld.Loader, root ipld.Link, s selector.Selector, fn traversal.AdvVisitFn) error {
-	builder := defaultChooser(root, ipld.LinkContext{})
+func Traverse(ctx context.Context, loader ipld.Loader, chooser traversal.NodeBuilderChooser, root ipld.Link, s selector.Selector, fn traversal.AdvVisitFn) error {
```

Comment:
This function is rapidly reaching "too many arguments". Consider switching to functional arguments or some form of builder.

Reply:
getting there huh!